src/Cake.Common/Build/TFBuild/TFBuildCommands.cs

```diff
@@ -280,6 +280,22 @@ public void UploadArtifact(string folderName, FilePath file, string artifactName
             }, file.MakeAbsolute(_environment).FullPath);
         }
 
+        /// <inheritdoc />
+        public void UploadArtifactDirectory(DirectoryPath directory)
+        {
+            UploadArtifactDirectory(directory, directory.GetDirectoryName());
```

Comment:
Should guard `directory` parameter for null.
```csharp
 if (directory== null)
{
    throw new ArgumentNullException(nameof(directory));
}
```

Reply:
@devlead Will do. I was going to put them but I noticed none of the other methods had them. Would you want another PR to fix up the rest of them also?

Reply:
@gtbuchanan  yes, please that would be great.